### PR TITLE
[METABRIDGE-843] chore: disable blocked tokens

### PR DESCRIPTION
**Pull request by micaelae**
## **Description**
Changes
* adds utilities for fetching blockedTokens from the swaps api
* disables a dest token if it's in the blockedTokens list AND if it's not the same as the src token

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24591?quickstart=1)

## **Related issues**

Fixes: N/A

## **Manual testing steps**

1. Go to send page, chainId=1
2. ETH -> BNB should not be possible
3. BNB -> BNB should be possible
4. BNB -> ETH should not be possible
5. BNB -> USDC should not be possible
6. BNB -> TKN should not be possible

1. Switch to a chain that's not supported by swaps
2. There should be no errors/crashes when attempting to do a simple send flow
3. Console/Network tab should have no errors regarding the blockedTokens api call

## **Screenshots/Recordings**

Blocked dest tokens disabled:

https://github.com/MetaMask/metamask-extension/assets/100321200/cf9afe8f-25e1-4dda-9260-a7c04554b8d3

Disable swap+send when src token is blocked:

Screenshot 2024-05-21 at 4 38 04 PM

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/ducks/send/send.js [hunk 1]
```diff
@@ -511,6 +514,7 @@ export const initialState = {
     balance: '0x0',
   },
   stage: SEND_STAGES.INACTIVE,
+  swapsBlockedTokens: [],
```
Comment (BZahory) [inline ui/ducks/send/send.js:517]: Should this be an object?

**Comment by BZahory** [comment]
You could update the `isSwapAllowed` in `ui/components/multichain/pages/send/components/recipient-content.tsx` to block swaps when a blocked token is selected on send

File: ui/ducks/send/send.js (hunk 1 above)
Comment (micaelae) [reply ui/ducks/send/send.js:517]: storing the `blockedTokens` responses per chain can add bloat to state and I don't see a reason why we would need access to multiple chain lists at once (yet). the api responses are cached for 5 mins in the browser so the extension won't be hitting the endpoint each time the page loads or the user switches chains anyways

Comment (BZahory) [reply ui/ducks/send/send.js:517]: Fair enough

File: ui/components/multichain/asset-picker-amount/asset-picker/asset-picker.tsx [hunk 2]
```diff
@@ -41,13 +41,15 @@ export type AssetPickerProps = {
...
   sendingAsset?: Asset;
+  isDisabled?: boolean;
 };
 
 // A component that lets the user pick from a list of assets.
 export function AssetPicker({
   asset,
   onAssetChange,
   sendingAsset,
+  isDisabled = false,
```
Comment (BZahory) [inline ui/components/multichain/asset-picker-amount/asset-picker/asset-picker.tsx:52]: Should we outright disable it? The current functionality shows the "send" option for both when "disabled"

**Review by ejwessel — APPROVED** [review]

**Review by BZahory — APPROVED** [review]

**Pull request merged**